lib/account-data.js

```diff
@@ -1,95 +1,122 @@
+        if(!error2.success){
+            return error.FOLDER_DELETE_ERROR
+        }
 
-}
+    }
+    if(nickname === global_state.config.currentUser){
+        const error3 =  await removeParsecUserFile()
+        if(error3){
+            return error3
+        }
+        if(global_state.data.keys().length === 0){
+            global_state.config.currentUser = ""
+        }
+        else {
+            global_state.config.currentUser = global_state.data.keys()[1]
+        }
+    }
+    delete global_state.data[nickname]
 
-function createAccount(nickname){
+    return error.SUCCESS
 
 
-    if(!fs.existsSync(multiUserFolder)){
-        fs.mkdirSync(multiUserFolder)
-    }
-    let newUserFolder = path.join(multiUserFolder,nickname)
-    let newUserFile = path.join(newUserFolder,'user.bin')
-    fs.mkdirSync(newUserFolder)
-    fs.copyFileSync(parsecUser,newUserFile)
 
-    data[nickname] = newUserFile
 }
+async function switchAccountData(nickname){
+    let parsecUser = global_state.config['parsecUser']
+
+    if(global_state.config.currentUser){
+        let currentUserDirectory = path.join(constants.multiUserPath,global_state.config.currentUser)
+        let currentUserFilePath = path.join(currentUserDirectory,'user.bin')
+
+        if(await fileFuncs.checkExists(parsecUser)){
+            let op = await fileFuncs.moveFile(parsecUser,currentUserFilePath)
+            if(!op.success){
+                return error.MOVE_FILE_ERROR
+            }
+        }
+    }
 
-function removeAccount(nickname){
+    let nicknameFolder = path.join(constants.multiUserPath,nickname)
+    let nicknameFile = path.join(nicknameFolder,'user.bin')
 
-    delete data[nickname]
-    let userPathWithoutBin = path.join(multiUserFolder,nickname)
-    let userPath = path.join(multiUserFolder,nickname,'user.bin')
-    if(fs.existsSync(userPath)){
-        rimraf.sync(userPath)
-        rimraf.sync(userPathWithoutBin)
-    }
 
 
 
-}
-function switchAccountData(nickname){
+    if(! await fileFuncs.checkExists(nicknameFolder)){
+        let op = await fileFuncs.makeDir(nicknameFolder)
+        if(!op.success){
+            return error.MAKE_DIR_ERROR
+        }
+    }
+    if(await fileFuncs.checkExists(nicknameFile)){
+        let op = await fileFuncs.moveFile(nicknameFile,parsecUser)
+        if(!op.success){
+            return error.MOVE_FILE_ERROR
+        }
+    }
+    if(!global_state.data[nickname]){
+        global_state.data[nickname] = nicknameFile
+    }
+
 
-    let userData = data[nickname]
-    rimraf.sync(parsecUser)
-    fs.copyFileSync(userData,parsecUser)
-    fs.writeFileSync(currentAccountFile, nickname)
+    global_state.config.currentUser = nickname
+    return error.SUCCESS
 }
-function checkUserFile(){
-    if (fs.existsSync(parsecUser)){
+async function checkUserFile(){
+    if (await fileFuncs.checkExists(global_state.config['parsecUser'])){
         return 1
     }
-    else return 0
+    else return error.SUCCESS
```

Comment:
no need to write else as it'll already return before, or you could even use ternary operator

also use error.REASON instead of directly returning 1 will be more readable

Reply:
done